### PR TITLE
Reprocess submissions

### PR DESCRIPTION
**Pull request by dcdarrell9**
## Checklist
  - [ ] CHANGELOG.md updated? (if required)

### CODE REVIEW

**Comment by dcdarrell9** [comment]
[SDX Console 2.0 - Reprocess submissions](https://trello.com/c/ZmnWY21Q/447-sdx-console-20-reprocess-submissions)

File: console/templates/store.html [hunk 1]
```diff
@@ -69,8 +70,8 @@
             <td class="table__cell">{{ item.metadata.user_id }}</td>
             <td class="table__cell">{{ item.submitted_at }}</td>
             <td class="table__cell">{{ item.invalid }}</td>
-            <input type="hidden" name="json_data" value="{{ item }}">
-            <td align="center">
+            <input id="json_data" type="hidden" name="json_data" value="{{ item }}">
```
Comment (warrenbailey) [inline console/templates/store.html]: Why submit the whole json data when you only need the tx_id?

I'd be tempted to do something like this:
'''input type="hidden" id="tx_id" name="tx_id" value="{{item.tx_id}}"'''

File: console/views/store.py [hunk 2]
```diff
@@ -65,62 +65,64 @@ def get_filtered_responses(logger, valid, tx_id, ru_ref, survey_id, datetime_ear
...
+    )
+
+    publisher.publish_message(tx_id, headers={'tx_id': tx_id})
+
+
+@store_bp.route('/reprocess', strict_slashes=False, methods=['POST'])
+@flask_security.login_required
+def reprocess_submission():
+    tx_id = str(request.get_data())[1:]
+    tx_id = tx_id.replace("'", "")
```
Comment (warrenbailey) [inline console/views/store.py]: Then this bit becomes much simpler:
   
tx_id = request.form.get('tx_id'')

File: console/views/store.py [hunk 3]
```diff
@@ -65,62 +65,64 @@ def get_filtered_responses(logger, valid, tx_id, ru_ref, survey_id, datetime_ear
...
+    return "Data reprocessed", 200
 
 
 @store_bp.route('/store', strict_slashes=False, defaults={'page': 0}, methods=['GET', 'POST'])
 @store_bp.route('/store/<page>', strict_slashes=False, methods=['GET', 'POST'])
 @flask_security.login_required
 def store(page):
     audited_logger = logger.bind(user=flask_security.core.current_user.email)
+
+    valid = request.args.get('valid', type=str, default='')
```
Comment (warrenbailey) [inline console/views/store.py]: Did this need to move?

Seems a little odd to do this if we're dealing with a post as request.args are query string parameters

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: console/views/store.py (hunk 3 above)
Comment (dcdarrell9) [reply console/views/store.py]: This is where we thought there was an issue with the post request, if we remove it then it breaks

Comment (AndrewTorrance) [reply console/views/store.py]: I think we need to get to the bottom of this before we merge . Hopefully I will get a chance to look again later

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**